mlir/include/mlir/IR/OpImplementation.h

```diff
@@ -1820,7 +1824,9 @@ ParseResult parseDimensionList(OpAsmParser &parser,
 //===--------------------------------------------------------------------===//
 
 /// The OpAsmOpInterface, see OpAsmInterface.td for more details.
-#include "mlir/IR/OpAsmInterface.h.inc"
+#include "mlir/IR/OpAsmTypeInterface.h.inc"
+// put Attr/Type before Op
```

Comment:
Could you document why this ordering is needed? And are there any ordering requirements user side?

Reply:
This was to prevent clang format from automatically sort them where the order becomes `Op->Type` but op depends on type. Now this restriction no longer exist, I'll clean it.